src/character_armor.cpp

```diff
@@ -119,6 +119,7 @@ static void armor_enchantment_adjust( Character &guy, damage_unit &du )
     } else if( du.type == STATIC( damage_type_id( "bullet" ) ) ) {
         du.amount = guy.calculate_by_enchantment( du.amount, enchant_vals::mod::ARMOR_BULLET );
     }
+    du.amount = guy.calculate_by_enchantment( du.amount, enchant_vals::mod::ARMOR_ALL );
```

Comment:
This breaks the `pure` damage type (mostly found in magiclysm). Pure is meant to never be reduced.

Recommend you check if damage type is NOT pure and then apply ARMOR_ALL's reduction.

Reply:
I forgot about pure damage.  Added a check to not add ARMOR_ALL if its pure damage.